datafn/python/datafn/idempotency.py

```
@@ -0,0 +1,54 @@
+        )
+        if record:
+            result = record.get("result")
+            if result:
+                result["deduped"] = True
+                return result
+    except Exception:
+        # If DB error, treat as cache miss (safer than blocking, but arguably less safe for idempotency)
+        # Consistent with "best effort" if DB is flaky, but if DB is down, everything is down.
+        # Letting error propagate might be better? 
+        # For now, swallow to match "cache miss" behavior unless critical.
+        pass
```

Ruff (0.15.7)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)